### PR TITLE
MM-25516: Changed to byte slice instead of string for cluster messages

### PR DESCRIPTION
**Pull request by agnivade**
https://mattermost.atlassian.net/browse/MM-25116

Testing:
Manually tested.
Load-tested with Cluster Controller.

I looked into changing the serialization method to use msgpack,
but the ClusterMessage struct was mainly used for only 3 fields
which didn't lead to much of a CPU time improvement, whereas
actually led to more allocations using msgpack. Hence, I chose
to remain with JSON.

```
name              old time/op    new time/op    delta
ClusterMarshal-8    3.51µs ± 1%    3.10µs ± 2%  -11.59%  (p=0.000 n=9+10)

name              old alloc/op   new alloc/op   delta
ClusterMarshal-8      776B ± 0%     1000B ± 0%  +28.87%  (p=0.000 n=10+10)

name              old allocs/op  new allocs/op  delta
ClusterMarshal-8      12.0 ± 0%      13.0 ± 0%   +8.33%  (p=0.000 n=10+10)
```

```release-note
Changed the field type of Data in model.ClusterMessage to []byte from string.
```

### CODE REVIEW

File: app/busy.go [hunk 1]
```diff
@@ -109,11 +110,12 @@ func (b *Busy) notifyServerBusyChange(sbs *model.ServerBusyState) {
 	if b.cluster == nil {
 		return
 	}
+	buf, _ := json.Marshal(sbs)
```
Comment (agnivade) [inline app/busy.go:113]: I didn't want to check the error just yet. Kept the scope of the PR just to changing the field type.

File: app/plugin_event.go [hunk 2]
```diff
@@ -4,16 +4,19 @@
 package app
 
 import (
+	"encoding/json"
+
 	"github.com/mattermost/mattermost-server/v6/model"
 )
 
 func (s *Server) notifyClusterPluginEvent(event model.ClusterEvent, data model.PluginEventData) {
+	buf, _ := json.Marshal(data)
```
Comment (agnivade) [inline app/plugin_event.go:13]: Same

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

File: app/busy.go (hunk 1 above)
Comment (streamer45) [reply app/busy.go:113]: Let's just make sure to track this.

Comment (agnivade) [reply app/busy.go:113]: https://mattermost.atlassian.net/browse/MM-37409

**Comment by agnivade** [comment]
@amyblais - Just a quite note, this PR requires an upgrade note for v6.

**Pull request merged**